pytorch_lightning/trainer/supporters.py

```diff
@@ -457,6 +457,23 @@ def _wrap_loaders_max_size_cycle(self) -> Any:
             )
             state.reset()
 
+    def _apply_cycle_iterator_length(self) -> None:
+        """When the model is `max_size_cycle`, compute the length across all ``CycleIterator`` and re-assign it to
+        all dataloaders."""
+        if self.mode != "max_size_cycle":
+            return
+
+        all_lengths = apply_to_collection(
+            self.loaders, CycleIterator, lambda c: get_len(c.loader), wrong_dtype=(Sequence, Mapping)
+        )
+        length = _nested_calc_num_data(all_lengths, max)
+
+        def _apply_fn(cycle_iterator: CycleIterator) -> None:
+            nonlocal length
```

Comment:
This could take the value to avoid the nonlocal

```suggestion
        def _apply_fn(cycle_iterator: CycleIterator, length: int) -> None:
```

Reply:
Done !